cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/LineageWriterDatasetFramework.java

```diff
@@ -125,12 +128,21 @@ public boolean hasType(Id.DatasetType datasetTypeId) throws DatasetManagementExc
   @Override
   public void deleteInstance(Id.DatasetInstance datasetInstanceId)
     throws DatasetManagementException, IOException, ServiceUnavailableException {
+    // Remove metadata for the dataset
+    businessMds.removeMetadata(datasetInstanceId);
```

Comment:
it would be better if these operations happened in the same transaction.  Guessing it's too late now, but do we have a jira open for it?